### ISSUE TITLE
invalid_token using refresh tokens in the same second

### ISSUE THREAD

**Issue by gm-ghanover** [tymondesigns/jwt-auth#1087]
JTI is an md5 of 'sub' and 'nbf'. The problem is if you have a user that makes multiple requests in 1 second, they have the same 'nbf' timestamp, so you end up returning a refresh token that is blacklisted (blacklist being based on JTI). 

\Tymon\JWTAuth\PayloadFactory::jti() should return something more unique, either by changing 'nbf' to include microseconds or simply using a randomly generated guid.

**Comment by gm-ghanover** [comment]
looks like this is already fixed in develop (via b3c7c991be39c9402d35b518332ce883b4948167, later updated with 965df4dc6ed2670d8512acaacf67ad2cb47c0472). is it possible to get a 0.5.10 build (we will never get a beta build approved for production release) with just this change?

```
diff --git a/src/PayloadFactory.php b/src/PayloadFactory.php
index d15acfb..c7784c8 100644
--- a/src/PayloadFactory.php
+++ b/src/PayloadFactory.php
@@ -12,6 +12,7 @@
 namespace Tymon\JWTAuth;

 use Illuminate\Http\Request;
+use Illuminate\Support\Str;
 use Tymon\JWTAuth\Claims\Factory;
 use Tymon\JWTAuth\Validators\PayloadValidator;

@@ -188,14 +189,7 @@ class PayloadFactory
      */
     protected function jti()
     {
-        $sub = array_get($this->claims, 'sub', '');
-        $nbf = array_get($this->claims, 'nbf', '');
-
-        if (! is_string($sub)) {
-            $sub = json_encode($sub);
-        }
-
-        return md5(sprintf('jti.%s.%s', $sub, $nbf));
+        return Str::random();
     }

     /**

```

**Comment by mikemillers** [comment]
Just out of interest as the JTI is supposed to be a unique ID what is wrong with using:

    protected function jti()
    {
         return uniqid(Str::random(),true);
    }

This seems like a stronger guarantee of uniqueness when generating a lot of tokens. This is how I had fixed the issue in my code.

**Comment by gm-ghanover** [comment]
Str::random() uses openssl_pseudo_random_bytes, if I recall

**Comment by mikemillers** [comment]
Yup you recall correctly. A collision is unlikely I know but as uniqid is based on microtime, with the random string prepended, plus the additional entropy option passed.. you see where I am going with that. My concern is not uniqueness of the tokens (the timestamp claims etc result in similar outcome for token uniqueness) but of the JTI's themselves. It appears similar thinking led to the original JTI being a product of the nbf in the first place. Uniqid is closer to that approach than the shift to just a random string
